geode-gfsh/src/main/java/org/apache/geode/management/internal/cli/commands/StopGatewaySenderCommandDelegateParallelImpl.java

```diff
@@ -0,0 +1,111 @@
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Future;
+
+import org.apache.geode.annotations.VisibleForTesting;
+import org.apache.geode.cache.Cache;
+import org.apache.geode.distributed.DistributedMember;
+import org.apache.geode.logging.internal.executors.LoggingExecutors;
+import org.apache.geode.management.internal.SystemManagementService;
+import org.apache.geode.management.internal.cli.result.model.ResultModel;
+import org.apache.geode.management.internal.cli.result.model.TabularResultModel;
+import org.apache.geode.management.internal.i18n.CliStrings;
+
+class StopGatewaySenderCommandDelegateParallelImpl
+    implements StopGatewaySenderCommandDelegate {
+
+  private final ExecutorService executorService;
+  private final SystemManagementService managementService;
+  private final StopGatewaySenderOnMemberFactory stopperOnMemberFactory;
+
+  @VisibleForTesting
+  public StopGatewaySenderCommandDelegateParallelImpl(ExecutorService executorService,
+      SystemManagementService managementService,
+      StopGatewaySenderOnMemberFactory stopperOnMemberFactory) {
+    this.executorService = executorService;
+    this.managementService = managementService;
+    this.stopperOnMemberFactory = stopperOnMemberFactory;
+  }
+
+  public StopGatewaySenderCommandDelegateParallelImpl(SystemManagementService managementService) {
+    executorService = LoggingExecutors.newCachedThreadPool("Stop Sender Command Thread ", true);;
+    stopperOnMemberFactory = new StopGatewaySenderOnMemberFactoryWithBeanImpl();
+    this.managementService = managementService;
+  }
+
+  public ResultModel executeStopGatewaySender(String id, Cache cache,
+      Set<DistributedMember> dsMembers) {
+    List<Callable<List<String>>> callables = new ArrayList<>();
+    for (final DistributedMember member : dsMembers) {
+      callables.add(() -> stopperOnMemberFactory.create().executeStopGatewaySenderOnMember(id,
+          cache, managementService, member));
+    }
+
+    List<Future<List<String>>> futures;
+    try {
+      futures = executorService.invokeAll(callables);
+    } catch (InterruptedException ite) {
+      Thread.currentThread().interrupt();
+      return ResultModel.createError(
+          CliStrings.format(CliStrings.GATEWAY_SENDER_STOP_0_COULD_NOT_BE_INVOKED_DUE_TO_1, id,
+              ite.getMessage()));
+    } finally {
+      executorService.shutdown();
+    }
+
+    return buildResultModelFromMembersResponses(id, dsMembers, futures);
+  }
+
+  private ResultModel buildResultModelFromMembersResponses(String id,
+      Set<DistributedMember> dsMembers, List<Future<List<String>>> futures) {
+    ResultModel resultModel = new ResultModel();
+    TabularResultModel resultData = resultModel.addTable(CliStrings.STOP_GATEWAYSENDER);
+    Iterator<DistributedMember> memberIterator = dsMembers.iterator();
+    for (Future<List<String>> future : futures) {
+      DistributedMember member = memberIterator.next();
```

Comment:
I have created a `List` from the `dsMembers` `Set` and I have passed it to the method so that we can be sure that the order between the `dsMembers` and the `futures` is the same.